python/tank/platform/interactive_authentication.py

```diff
@@ -71,15 +71,19 @@ class AuthenticationHandlerBase(object):
     be impossible to authenticate again.
     """
 
-    def authenticate(self):
+    def authenticate(self, force_human_user_authentication):
         """
         Common login logic, regardless of how we are actually logging in. It will first try to reuse
         any existing session and if that fails then it will ask for credentials and upon success
         the credentials will be cached.
+        :param force_human_user_authentication: Indicates if we should force authentication to be user based.
```

Comment:
Would be nice with a `:raises:` here since this method raises special exceptions!